packages/babel-helpers/src/helpers.js

```diff
@@ -425,6 +425,47 @@ helpers.inheritsLoose = defineHelper(`
   }
 `);
 
+// Based on https://github.com/WebReflection/babel-plugin-transform-builtin-classes
+helpers.wrapNativeSuper = defineHelper(`
+  var _gPO = Object.getPrototypeOf || function _gPO(o) { return o.__proto__ };
+  var _sPO = Object.setPrototypeOf || function _sPO(o, p) { o.__proto__ = p };
+  var _construct = (typeof Reflect === "object" && Reflect.construct) ||
+    function _construct(Parent, args, Class) {
+      var Constructor, a = [null];
+      a.push.apply(a, args);
+      Constructor = Parent.bind.apply(Parent, a);
+      return _sPO(new Constructor, Class.prototype);
+    };
+
+  var _cache = typeof Map === "function" && new Map();
+
+  export default function _wrapNativeSuper(Class) {
+    if (typeof _cache !== "undefined") {
```

Comment:
It'd be really nice if this didn't wrap emulated classes which already have callable constructors, like when using the Custom Elements polyfill. In that case `HTMLElement` is callable because the polyfill wraps the native class. Detecting this case would avoid double-wrapping.

It's unfortunately impossible to absolutely detect that a constructor function is callable, but there's an approximation that might be suitable here:

```js
function isNative(ctor) {
  return ctor.toString().indexOf('[native code]') !== -1;
}
```

Reply:
I don't think we need to catch user defined classes with a function called `isNative` ...

Reply:
It wouldn't work, because core-js makes methods "look native". e.g.
https://github.com/zloirock/core-js/blob/16357e4fefa070573a342ef2a5692f30ba4320c6/tests/tests/es6.promise.ls#L9
https://github.com/zloirock/core-js/blob/6e389278357a6296a58155742871ca876dff2bc3/tests/helpers/looks-native.ls

Reply:
there are here and there Babel specific conventions, like `__esModule = true`, I wonder if it's worth considering a non enumerable, non configurable, `__nativeWrap = true` to explicitly opt out from Babel wraps so that the check would be `builtins.includes(ctor) && !ctor.__nativeWrap`

Simple enough to cover polyfills case.

Reply:
@Kovensky:

> This would also catch user-defined classes, but one could then question whether you actually need this transform if you expect to find such a class

It happens, especially in the case where a library is published compiled to npm, and used by an app that's shipping native ES6. We try to discourage distributing compiled code, but we can't control that.

We could indeed add a flag to the polyfill. What we really want is this: https://esdiscuss.org/topic/add-reflect-isconstructor-and-reflect-iscallable

Reply:
I think that this discussion caan be deferred, since it *works* as it is implemented in this PR.
Don't wrap classess without necessity is a possible performance optimization, but for now shipping a working implementation is enough.